src/domain/user.js

```diff
@@ -236,12 +236,67 @@ export default class User {
     })
     return updatedUser
   }
+
+  static async searchUserByName(name) {
+    const splitName = name.split(' ')
+
+    const searchedUser = await dbClient.user.findMany({
+      where: {
+        OR: [
+          {
+            profile: {
+              firstName: {
+                contains: name,
+                mode: 'insensitive'
+              }
+            }
+          },
+          {
+            profile: {
+              lastName: {
+                contains: name,
+                mode: 'insensitive'
+              }
+            }
+          },
+          {
+            AND:
+              splitName.length > 1
+                ? [
+                    {
+                      profile: {
+                        firstName: {
+                          contains: splitName[0],
+                          mode: 'insensitive'
+                        }
+                      }
+                    },
+                    {
+                      profile: {
+                        lastName: {
+                          contains: splitName[1],
+                          mode: 'insensitive'
+                        }
+                      }
+                    }
+                  ]
+                : []
+          }
+        ]
+      },
+      include: {
+        profile: true,
+        cohort: true
+      }
+    })
+    return searchedUser.map((user) => User.fromDb(user))
+  }
 }
 
-async function _findByUniqueAsATeacher(key, value) {
+async function _findByUniqueAsATeacher(id) {
```

Comment:
instead of id it should be [key]: value

imagine in future we want to find a user with his email address. if the Parameters be key and value, we can use the same function. but if the parameters be id, we need to create another function to the same thing